node/src/actors/chain_manager/actor.rs

```diff
@@ -305,6 +385,57 @@ impl ChainManager {
         Box::pin(fut)
     }
 
+    /// Ensure that there is only one ChainState in the storage. Old versions of witnet-rust used
+    /// to allow having multiple ChainStates for multiple testnets. This function will panic in that
+    /// case because multiple ChainStates are incompatible with storing UTXOs as keys in the
+    /// database.
+    pub fn check_only_one_chain_state_in_storage(&mut self, ctx: &mut Context<ChainManager>) {
+        let fut = config_mngr::get()
+            .into_actor(self)
+            .map_err(|err, _act, _ctx| {
+                log::error!("Couldn't get config: {}", err);
+            })
+            .and_then(|config, act, _ctx| {
+                storage_mngr::get_backend()
+                    .into_actor(act)
+                    .map_err(|err, _act, _ctx| {
+                        log::error!("Failed to get storage backend: {}", err);
+                    })
+                    .map_ok(move |backend, _act, _ctx| (config, backend))
+            })
+            .map_ok(|(config, backend), _act, _ctx| {
+                let magic = config.consensus_constants.get_magic();
+
+                let all_chain_states: Vec<_> = backend
+                    .prefix_iterator(b"chain-")
```

Comment:
This doesn't work because the chain state key is serialized using bincode, and the prefix includes the key length as a little endian u64. So we would need to check all the possible key lengths, from "chain-0" to "chain-65536".

Reply:
Fixed